### PR TITLE
[GEOS-9462] Downgrade ArcSDE plugin to an unsupported  module

### PR DESCRIPTION
**Pull request by groldan**
GeoTools side of the fence for https://github.com/geoserver/geoserver/pull/3988

## Checklist

> Reviewing is a process done by project maintainers, mostly on a volunteer basis. We try to keep the overhead as small as possible and appreciate if you help us to do so by completing the following items. Feel free to ask in a comment if you have troubles with any of them.

For all pull requests:

- [x] Confirm you have read the [contribution guidelines](https://github.com/geotools/geotools/blob/master/CONTRIBUTING.md) 
- [x] You have sent a Contribution Licence Agreement (CLA) as necessary (not required for small changes, e.g., fixing typos in documentation)
- [x] Make sure the first PR targets the master branch, eventual backports will be managed later. This can be ignored if the PR is fixing an issue that only happens in a specific branch, but not in newer ones.
- [ ] The changes are not breaking the build in downstream projects using SNAPSHOT dependencies, GeoWebCache and GeoServer.

The following are required only for core and extension modules (they are welcomed, but not required, for unsupported modules):
- [x] There is a ticket in Jira describing the issue/improvement/feature (a notable exemptions is, changes not visible to end users)
- [ ] PR for bug fixes and small new features are presented as a single commit
- [x] Commit message must be in the form "[GEOT-XYZW] Title of the Jira ticket"
- [ ] New unit tests have been added covering the changes
- [ ] This PR passes all existing unit tests (test results will be reported by travis-ci after opening this PR)
- [ ] This PR passes the [QA checks](https://docs.geotools.org/latest/developer/conventions/code/qa.html) (QA checks results will be reported by travis-ci after opening this PR)
- [x] Documentation has been updated accordingly.

Submitting the PR does not require you to check all items, but by the time it gets merged, they should be either satisfied or inapplicable.

### CODE REVIEW

File: docs/user/library/data/arcsde.rst [hunk 1]
```diff
@@ -1,7 +1,9 @@
 ArcSDE Plugin
 -------------
 
-The ArcSDE plugin allows the GeoTools developer to work with the `ArcSDE <http://en.wikipedia.org/wiki/ArcSDE>`_ middle ware. The ArcSDE plugin has been with us since GeoTools 2.0 and is much loved.
+Unsupported plugin allowing to work with the `ArcSDE <http://en.wikipedia.org/wiki/ArcSDE>`_ middle ware.
+The ArcSDE plugin has been with us since GeoTools 2.0 and is much loved, yet it has become unsupported
```
Comment (aaime) [inline docs/user/library/data/arcsde.rst]: A plugin with sentiment! (that must be a first)

File: modules/unsupported/arcsde/common/pom.xml [hunk 2]
```diff
@@ -123,7 +123,7 @@
       </build>
     </profile>
     <profile>
-      <id>arcsde</id>
+      <id>arcsde-jars</id>
```
Comment (aaime) [inline modules/unsupported/arcsde/common/pom.xml]: What happened here?

File: docs/user/library/data/arcsde.rst (hunk 1 above)
Comment (groldan) [reply docs/user/library/data/arcsde.rst]: lol, didn't want to change the existing statement :)

File: modules/unsupported/arcsde/common/pom.xml (hunk 2 above)
Comment (groldan) [reply modules/unsupported/arcsde/common/pom.xml]: not sure, let me double check.

Comment (groldan) [reply modules/unsupported/arcsde/common/pom.xml]: ok, that's actually a fix. That profile is meant for development when you do have the proprietary jars in your local maven repo. Both common/pom.xml and datastore/pom.xml use it, but in common it was called `arcsde` and in datastore `arcsde-jars`.

**Comment by aaime** [comment]
Looks good to me, @ianturton do you also want to have a look before merge? (you were also requested as a reviewer)

**Pull request merged**